client/my-sites/email/form/mailboxes/components/mailbox-form-wrapper/index.tsx

```diff
@@ -1,50 +1,36 @@
-import { Button } from '@automattic/components';
-import { translate, useRtl } from 'i18n-calypso';
-import { useState } from 'react';
+import { useRtl } from 'i18n-calypso';
 import { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
-import { MailboxField } from 'calypso/my-sites/email/form/mailboxes/components/field';
+import { MailboxField } from 'calypso/my-sites/email/form/mailboxes/components/mailbox-field';
 import {
 	EmailProvider,
-	FormFieldNames,
 	GoogleMailboxFormFields,
 	MailboxFormFieldBase,
 	TitanMailboxFormFields,
 } from 'calypso/my-sites/email/form/mailboxes/types';
 
 interface MailboxFormWrapperProps {
-	onSubmit: ( mailbox: MailboxForm< EmailProvider > ) => void;
-	provider: EmailProvider;
-	selectedDomainName: string;
+	mailbox: MailboxForm< EmailProvider >;
+	onFieldValueChanged?: ( field: MailboxFormFieldBase< string > ) => void;
 }
 
 const MailboxFormWrapper = ( {
 	children,
-	onSubmit,
-	provider,
-	selectedDomainName,
-}: MailboxFormWrapperProps & { children?: JSX.Element } ) => {
+	mailbox,
+	onFieldValueChanged = () => undefined,
+}: MailboxFormWrapperProps & {
+	children?: JSX.Element | false;
+} ): JSX.Element => {
 	const isRtl = useRtl();
-	const [ { mailbox }, setMailboxState ] = useState( {
-		mailbox: new MailboxForm< EmailProvider >( provider, selectedDomainName ),
-	} );
-
-	const renderedFields = new Set< string >();
-
-	const handleSubmit = () => {
-		renderedFields.forEach( ( fieldName ) => mailbox.validateField( fieldName as FormFieldNames ) );
-		setMailboxState( { mailbox } );
-		onSubmit( mailbox );
-	};
 
 	const formFields = mailbox.formFields;
 	const domainAffix = {
 		[ isRtl ? 'textInputPrefix' : 'textInputSuffix' ]: `\u200e@${ formFields.domain.value }\u202c`,
 	};
 
 	const commonProps = ( field: MailboxFormFieldBase< string > ) => {
-		renderedFields.add( field.fieldName );
 		return {
 			field,
```

Comment:
Maybe we should add `disabled: areButtonsBusy` here (and pass that prop into this component as well). Should prevent scenarios where users edit the form fields during submission.